sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp

```diff
@@ -57,29 +57,30 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
      */
     Azure::Core::Response<Models::AcquireShareLeaseResult> Acquire(
         std::chrono::seconds duration,
-        const AcquireShareLeaseOptions& options = AcquireShareLeaseOptions());
+        const AcquireShareLeaseOptions& options = AcquireShareLeaseOptions()) const;
 
     /**
      * @brief Releases the file or share's previously-acquired lease.
      *
      * @param leaseId ID of the previously-acquired lease.
      * @param options Optional parameters to execute this function.
-     * @return A ReleaseShareLeaseResult describing the updated container.
+     * @return A ReleaseShareLeaseResult describing the updated share or file.
      */
     Azure::Core::Response<Models::ReleaseShareLeaseResult> Release(
-        const ReleaseShareLeaseOptions& options = ReleaseShareLeaseOptions());
+        const ReleaseShareLeaseOptions& options = ReleaseShareLeaseOptions()) const;
 
     /**
      * @brief Changes the lease of an active lease.
      *
      * @param leaseId ID of the previously-acquired lease.
      * @param proposedLeaseId Proposed lease ID, in a GUID string format.
      * @param options Optional parameters to execute this function.
-     * @return A ChangeShareLeaseResult describing the lease.
+     * @return A ChangeShareLeaseResult describing the updated lease.
+     * @remarks The current ShareLeaseClient becomes invalid if this operation succeeds.
```

Comment:
Not only if the operation succeeds, but also when the operation fails for some particular failures, including the response is not received for a successful result or if the server times out but the change is done before the timeout.
How about `The current ShareLeaseClient becomes invalid or undefined state after this function is called.`

Reply:
I believe it's a general rule that the result is undetermined (meaning we cannot assume the result being successful or failed) if errors like network failure happens